### PR TITLE
FE-413: Improve Sentry issue filtering, tags

### PR DESCRIPTION
**Pull request by bkemper**
What to test?

- Confirm render errors are reported to Sentry
- Confirm every reported error was a documentLanguage, navigatorLanguage, and source
- Confirm browser extension errors are reported to Sentry as warnings
- Confirm API errors are reported as warnings
- Confirm unable to load application errors are reported to Sentry as warnings

### CODE REVIEW

**Comment by ewandennis** [comment]
- [x] "unable to load" issues are reported as warnings
 - [ ] Reports contain `documentLanguage`, `navigatorLanguage`, and `source`
 - [ ] Render errors are reported to Sentry
 - [ ] Browser extension errors are reported as warnings
 - [ ] SparkPost and Zuora API errors are reported as warnings

File: src/components/errorBoundaries/ErrorBoundary.js [hunk 1]
```diff
@@ -15,12 +14,11 @@ export default class ErrorBoundary extends Component {
     hasError: false
   }
 
-  componentDidCatch (error, info) {
+  componentDidCatch() {
     this.setState({ hasError: true });
-    ErrorTracker.report({ info }, error);
```
Comment (ewandennis) [inline src/components/errorBoundaries/ErrorBoundary.js]: How does errors during rendering flow to Sentry now, given that this component catches and doesn't report?

File: src/helpers/errorTracker.js [hunk 2]
```diff
@@ -42,22 +39,53 @@ export function breadcrumbCallback (crumb) {
...
       },
       user
     };
   };
 }
 
+// Check if error event was thrown from our bundle or from something else (i.e. browser extension)
+export function isErrorFromOurBundle(data) {
+  // The local environment match is looser to allow for hot module replacement (i.e. http://app.sparkpost.test/4.a0803f8355f692de1382.hot-update.js)
+  const looksLikeOurBundle = new RegExp('sparkpost.test/|sparkpost.com/static/js/');
```
[the remark this answers is not in the document]
Comment (bkemper) [reply src/helpers/errorTracker.js]: I relaxed and escaped this regex, but you should need to specific the port.  Hamurai should have setup this proxy_pass.

https://github.com/SparkPost/hamurai/blob/master/roles/sparkpost.openresty/templates/sparkpost.conf.j2#L122-L143

File: src/helpers/errorTracker.js [hunk 3]
```diff
@@ -42,22 +39,53 @@ export function breadcrumbCallback (crumb) {
...
+  // There should never be multiple exception values
+  let frames = _.get(data, 'exception.values[0].stacktrace.frames', []);
+  const firstFunction = _.get(frames, '[0].function');
+
+  // A Sentry function may be included in the stacktrace and needs to be ignored (i.e. HTMLDocument.wrapped or wrapped)
+  if (/wrapped/.test(firstFunction)) {
```
[the remark this answers is not in the document]
Comment (bkemper) [reply src/helpers/errorTracker.js]: It primarily happens with chrome extensions.  Try Ferdinand.

https://gist.github.com/bkemper/4c1e9a87f4b0b921711ea6791c265cc0

File: src/components/errorBoundaries/ErrorBoundary.js (hunk 1 above)
Comment (bkemper) [reply src/components/errorBoundaries/ErrorBoundary.js]: `componentDidCatch` is a misleading name.  It doesn't catch render errors like a try/catch.  It just listens for render errors and provides this convenient lifecycle method to recover.  Raven's onerror listener will still report render errors without this manually report.

**Comment by bkemper** [comment]
@ewandennis I think I answered all feedback.  I'll ping you first thing when I log on tomorrow to see if you have any questions or concerns.

File: src/helpers/tests/errorTracker.test.js [hunk 4]
```diff
@@ -16,28 +17,191 @@ describe('.breadcrumbCallback', () => {
...
+    expect(isErrorFromOurBundle(data)).toEqual(true);
+  });
+
+  it('returns true with error from our bundle when running locally', () => {
+    const data = {
+      exception: {
+        values: [{
+          stacktrace: {
+            frames: [
+              { filename: 'http://app.sparkpost.test/4.a0803f8355f692de1382.hot-update.js', function: 'render' }
```
Comment (ewandennis) [inline src/helpers/tests/errorTracker.test.js]: This is ok but it doesn't include a port number on the URL so it doesn't completely exercise our local environment detection.

**Pull request merged**